File: 11-container-with-most-water/11-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int max = 0;
        int left = 0, right = height.length - 1;
        
        while (left < right) {
            int area = Math.min(height[left], height[right]) * (right - left);
            max = Math.max(area, max);
            if (height[left] < height[right])
                left++;
            else
                right--;
        }
        return max;
    }
}

File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int len = height.size();
        int left = 0, right = len - 1;
        int ma = min(height[left], height[right]) * (right - left);
        while (left < right) {
            if (height[left] <= height[right])
                left++;
            else
                right--;
            
            ma = max(ma, min(height[left], height[right]) * (right - left));
        }
        return ma;
    }
};